### ISSUE TITLE
[BUG] tailscale_up_skip does not work as expected

### ISSUE THREAD

**Issue by chanux** [artis3n/ansible-role-tailscale#81]
**Describe the bug**
When setting `tailscale_up_skip` to true similar to the example in the readme (see below), the playbook fails if `tailscale_auth_key` is not set. I'm not sure if this is the intended behavior.

```
- name: Servers
  hosts: all
  roles:
    - role: artis3n.tailscale
      vars:
        tailscale_up_skip: true
```
I just set a fake value to `tailscale_auth_key` and the playbook goes through but gets stuck at the task "Bring Tailscale Up".

**Expected behavior**
Just install Tailscale when `tailscale_up_skip`

**Target (please complete the following information):**
 - OS: Debian Buster
 - Ansible version: 3.0.0
 - `artis3n.tailscale` version: v1.9.0
 - Tailscale version (set `verbose` to true):

Output of `tailscale status` during role execution (set `verbose` to true):

verbose does not seem to work.

**Additional context**

I'm using Ansible in Packer to create a GCP image.

Packer version: 1.7.0
Python version: 3.9.1

**Comment by artis3n** [comment]
Thanks for this issue! I will say I use `1.9.0` [here](https://github.com/artis3n/personal-wiki-infra/blob/main/packer/ansible/main.yml#L15) so I feel confident the `tailscale_up_skip` command does work.

Let's rule out some basics. How are you invoking the role in your task? `import_role`? Can you attach a snippet? Can you attach the output that comes from running this role? Can you run ansible with `-vv` for ansible's verbose output and attach the output of the "Bring Tailscale Up" command that hangs?

The `verbose` flag should output additional information right before this task. Do the status and version get outputted? No need to submit the output of "Tailscale Status" but would be helpful to attach "Tailscale Version" to confirm you're definitely using 1.9.0.

```yml
- name: Tailscale Status
  debug:
    var: tailscale_status
  when: verbose | bool

- name: Record Tailscale Version
  command: tailscale version
  changed_when: false
  register: tailscale_version
  when: verbose | bool

- name: Tailscale Version
  debug:
    msg: "{{ tailscale_version.stdout.split('\n') }}"
  when: verbose | bool

- name: Bring Tailscale Up
  become: yes
  # The command module cannot use | ; &
  # So we are ok not quoting the variables
  command: tailscale up --authkey={{ tailscale_auth_key }} {{ tailscale_args | default() }}
  # Since the auth key is included in this task, we do not want to log output
  no_log: true
  register: tailscale_start
  when: >
    not tailscale_up_skip | bool
    and '"hello.ipn.dev"' not in tailscale_status.stdout
  notify: Confirm Tailscale is Connected
```

If `verbose` is also not working for you, my first guess is you are actually on an older version. Run `ansible-galaxy role install --force -r requirements.yml` or `ansible-galaxy role install --force artis3n.tailscale` to make sure you overwrite your local cache and get the latest version. I tend to [always do that](https://github.com/artis3n/cloud-hackbox/blob/main/Makefile#L9) to avoid version issues.

**Comment by chanux** [comment]
Ansible said it's using `1.9.0` so I trusted. Anyhow, here's the output before and after setting  `galaxy install --force`

**Before**
```
==> googlecompute.tailscale-base: Provisioning with Ansible...
    googlecompute.tailscale-base: Setting up proxy adapter for Ansible....
    googlecompute.tailscale-base: Executing Ansible Galaxy
    googlecompute.tailscale-base: Starting galaxy role install process
    googlecompute.tailscale-base: [WARNING]: - artis3n.tailscale (v1.9.0) is already installed - use --force to
    googlecompute.tailscale-base: change version to unspecified
```
**After** (with `-vv` to Ansible)
```
==> googlecompute.tailscale-base: Provisioning with Ansible...
    googlecompute.tailscale-base: Setting up proxy adapter for Ansible....
    googlecompute.tailscale-base: Executing Ansible Galaxy
    googlecompute.tailscale-base: Starting galaxy role install process
    googlecompute.tailscale-base: - changing role artis3n.tailscale from v1.9.0 to unspecified
    googlecompute.tailscale-base: - downloading role 'tailscale', owned by artis3n
    googlecompute.tailscale-base: - downloading role from https://github.com/artis3n/ansible-role-tailscale/archive/v1.9.0.tar.gz
    googlecompute.tailscale-base: - extracting artis3n.tailscale to /Users/chanux/.ansible/roles/artis3n.tailscale
    googlecompute.tailscale-base: - artis3n.tailscale (v1.9.0) was installed successfully
```

I did eyeball the downloaded code and everything looks fine. It's just that the boolean values seem to be ignored.

I added the packer & co config I use in [this gist](https://gist.github.com/chanux/1cba0ef39942d9e440502164befb0fce). It's slightly modified in ways that should not affect the case we are testing (ex: I use a custom vpc/subnet)

**Comment by chanux** [comment]
I just tried out `import_role` from a task instead of directly using the role in the play. And it works!

Thank you very much for the prompt and detailed response.

**Comment by artis3n** [comment]
Absolutely! I got nervous when I saw your logs showing 1.9.0 was in use but didn't look into it further today.

I remember encountering that distinction between roles list and import_role in the past and honestly have no idea why variables behave differently. Nothing in the documentation suggests different behavior. The roles list is more "legacy" but both options are still supported. It seems it will remain a mystery.

Anyway, glad it's working for you!
